validator/rpc/standard_api.go

```diff
@@ -693,3 +696,43 @@ func validatePublicKey(pubkey []byte) error {
 	}
 	return nil
 }
+
+// SetVoluntaryExit TBD
+func (s *Server) SetVoluntaryExit(ctx context.Context, req *ethpbservice.SetVoluntaryExitRequest) (*ethpbservice.SetVoluntaryExitResponse, error) {
+	if s.validatorService == nil {
+		return nil, status.Error(codes.FailedPrecondition, "Validator service not ready")
+	}
+	if err := validatePublicKey(req.Pubkey); err != nil {
+		return nil, status.Error(codes.FailedPrecondition, err.Error())
+	}
+	if s.wallet == nil {
+		return nil, status.Error(codes.FailedPrecondition, "No wallet found")
+	}
+	km, err := s.validatorService.Keymanager()
+	if err != nil {
+		return nil, err
+	}
+	cfg := accounts.PerformExitCfg{
+		ValidatorClient: s.beaconNodeValidatorClient,
+		NodeClient:      s.beaconNodeClient,
+		Keymanager:      km,
+	}
+	sve, err := client.CreateSignedVoluntaryExit(ctx, cfg.ValidatorClient, cfg.NodeClient, cfg.Keymanager.Sign, req.Pubkey)
```

Comment:
you need to refactor this to take the req.epoch instead of using the current one for the signing if a query parameter is needed